sdc/rewrites/dataframe_getitem_attribute.py

```diff
@@ -0,0 +1,87 @@
+# *****************************************************************************
+# Copyright (c) 2020, Intel Corporation All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#
+#     Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+from numba.ir import Assign, Const, Expr, Var
+from numba.ir_utils import mk_unique_var
+from numba.rewrites import register_rewrite, Rewrite
+from numba.types import StringLiteral
+from numba.typing import signature
+
+from sdc.config import config_pipeline_hpat_default
+from sdc.hiframes.pd_dataframe_type import DataFrameType
+
+
+if not config_pipeline_hpat_default:
+    @register_rewrite('after-inference')
+    class RewriteDataFrameGetItemAttr(Rewrite):
+        """
+        Search for calls of df.attr and replace it with calls of df['attr']:
+        $0.2 = getattr(value=df, attr=A) -> $const0.0 = const(str, A)
+                                            $0.2 = static_getitem(value=df, index=A, index_var=$const0.0)
+        """
+
+        def match(self, func_ir, block, typemap, calltypes):
+            self.block = block
+            self.typemap = typemap
+            self.calltypes = calltypes
+            self.getattrs = getattrs = set()
+            for expr in block.find_exprs(op='getattr'):
+                obj = typemap[expr.value.name]
+                if not isinstance(obj, DataFrameType):
+                    continue
+                if expr.attr in obj.columns:
+                    getattrs.add(expr)
+
+            return len(getattrs) > 0
+
+        def apply(self):
+            new_block = self.block.copy()
+            new_block.clear()
```

Comment:
That's a bad idea. Block may contain a lot of functionality, including jumps and so on. You remove it all and add single statement.

Reply:
I add all statements from `self.block` to `new_block` additionally rewriting some instructions:
```python
for inst in self.block.body:
    # ...
    new_block.append(inst)
```

@AlexanderKalistratov could you confirm it should be fixed somehow?

Reply:
@densmirn you call `new_block.clear()` which erases all statements from original block. Isn't it?

Reply:
Ok, I get it.